app/assets/javascripts/components/components/status_content.jsx

```diff
@@ -93,7 +93,7 @@ const StatusContent = React.createClass({
     const { status } = this.props;
     const { hidden } = this.state;
 
-    const content = { __html: emojify(status.get('content')) };
+    const content = { __html: emojify(status.get('content').replace(/\s*(<p>|<\/p>)\s*/g, '$1').replace(/\s{2}/g, ' &nbsp;')) };
```

Comment:
Actually, the linked toot is a bad example since it uses full-width spaces which seem to be preserved regardless of any CSS whitespace settings.